wiki_articles/cp.md

```diff
@@ -1,38 +1,28 @@
-<!-- alias CP-help, CP -->
-
+<!-- alias cp-help, cp -->
 # The difficulty in helping with competitive coding problems
 
-Competitive programming problems are difficult to help with. The often require helpers to get very involved in problems, even taking the time to solve the problem themselves, and often also boil down to the person asking the question not having thought through the problem enough themselves to understand the complete.
-
-When asking a programming question, it's important to make the effort required for helpers as minimal as possible. 
-
-## How challenges are structured
-
-Competitive programming challenges often have arbitrary rules and requirements that require helpers to get deeply involved to even understand the problem.
-
-## Hidden test cases
+Please understand competitive programming (CP) questions are hard to help with:
 
-Test cases are often hidden to prevent, for example, hardcoding an edge case in your code rather then properly fixing the code. Without effort from the learner to find edge cases it's difficult for helpers to narrow down where issues occur.
+- Understanding requires getting into solving the question itself.
+- Test cases are hidden, so no starting point to find out what is wrong.
+- Your solution code is probably not written for others to easily read.
 
-## TLE (time limit exceeded)
-
-TLE failures are common and can be rather confusing for beginners. The question is often "how do I make this faster?". Often the reason for TLE is that your algorithm is not efficient enough, and it's often hard to know if your current algorithm is appropriate.
+<!-- inline -->
+## How to get the best possible help?
 
-## Common styling
+What you can do to get the best possible help:
 
-A very specific coding "style" has emerged in this specific programming niche. This is often full of hacks, aliases, 1-3 letter variables / acronyms, sparse or bad formatting, making code extremely difficult to parse for people to understand the code. And if they can't understand the code helping becomes even more difficult.
+- Format your code, this includes giving meaningful names to things.
+- Try to narrow down the problem, then ask a detailed, on point question.
```

Comment:
Avoid padding wording in wiki articles. Every character counts.
```suggestion
- Narrow down the problem, then ask a detailed, on-point question.
```
This might need further changes if we can save one line by removing `on-point`.